### DISCUSSION TITLE
problems launching `gatsby develop`

### DISCUSSION THREAD

**Discussion by IhorFedorov7** [gatsbyjs/gatsby#35313] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

by installing npm install -g gatsby-cli
after i created a new project gatsby new blog
I have a folder but without node_modules
when running gatsby develop gives an error and recommends installing manually npm install after repeated gatsby develop starts to give such errors

ERROR #98124 WEBPACK

Generating development SSR bundle failed

Can't resolve '@reach/router/lib/utils' in 'C:\Users\ihorf\Desktop\Igor\WORK\GATSBY\adlockBlog\.cache'

If you're trying to use a package make sure that '@reach/router/lib/utils' is installed. If you're trying to
use a local file to make sure that the path is correct.

File: .cache\find-path.js

 ERROR #98124 WEBPACK

Generating development SSR bundle failed

Can't resolve 'gatsby-link' in 'C:\Users\ihorf\Desktop\Igor\WORK\GATSBY\adlockBlog\.cache'
If you're trying to use a package make sure that 'gatsby-link' is installed. If you're trying to use a local
file make sure that the path is correct.

File: .cache\gatsby-browser-entry.js

 ERROR #98124 WEBPACK

Generating development SSR bundle failed

Can't resolve 'gatsby-react-router-scroll' in 'C:\Users\ihorf\Desktop\Igor\WORK\GATSBY\adlockBlog\.cache'

If you're trying to use a package make sure that 'gatsby-react-router-scroll' is installed. If you're trying
to use a local file make sure that the path is correct
![image_2022-03-31_09-20-56](https://user-images.githubusercontent.com/50741511/161000015-cc00aee6-2274-4598-91aa-db91686566c8.png)
.

### Reproduction Link

-

### Steps to Reproduce

1. npm install -g gatsby-cli
2. gatsby new blog
3. if there is no node_modules at the root of the folder
4. gatsby develop
5. npm install
6.  npm audit fix --force
7. npm audit fix
8. gatsby develop

### Expected Result

I'm waiting for help and solving the problem, because I really want to create a project on Gatsby js

### Actual Result

error

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (6) x64 Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz    
  Binaries:
    Node: 16.13.1 - C:\Program Files\nodejs\node.EXE        
    npm: 8.2.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.2
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (99.0.1150.55)
  npmPackages:
    gatsby: ^2.19.38 => 2.32.13
    gatsby-link: ^4.11.0 => 4.11.0
    gatsby-page-utils: ^2.11.0 => 2.11.0
    gatsby-plugin-gatsby-cloud: ^1.0.2 => 1.0.2
    gatsby-plugin-image: ^0.7.2 => 0.7.2
    gatsby-plugin-manifest: ^2.6.1 => 2.6.1
    gatsby-plugin-offline: ^3.10.2 => 3.10.2
    gatsby-plugin-react-helmet: ^3.10.0 => 3.10.0
    gatsby-plugin-sharp: ^2.14.4 => 2.14.4
    gatsby-react-router-scroll: ^5.11.0 => 5.11.0
    gatsby-source-filesystem: ^2.11.1 => 2.11.1
    gatsby-transformer-sharp: ^2.12.1 => 2.12.1
  npmGlobalPackages:
    gatsby: 4.11.0
```

**Comment by albergbl** [comment]
I'm having a similar issue after updating gatsby to 4.11.1 and running npm outdated and updating all to Wanted using npm. Terminal says: Module not found: Error: Can't resolve '@gatsbyjs/reach-router/lib/utils' in
'/Users/[myusername]/[foldername]/[gatsybyinstallation]/.cache' Names in brackets [] are descriptions only.

I then use npm to install reach/router via the command npm install @reach/router

The terminal says: 
"code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: gatsby-starter-default@0.1.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"15.x || 16.x || 16.4.0-alpha.0911da3" from @reach/router@1.3.4
npm ERR! node_modules/@reach/router
npm ERR!   @reach/router@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!"

I suspect these errors are caused by Gatsby 4.11.1 relying on react@18.0.0 and on @reach/router, which requires react@16.4.0-alpha.0911da3.

**Reply by tyhopp** [reply]
Hi @albergbl, Gatsby 4.11 doesn't require React 18 (you can use older versions) and uses a fork of reach router, so you might be running into issues by installing reach router again. Difficult to say more without more information but hopefully this bit helps.

If you do continue to encounter issues please create a new issue with a minimal reproduction. Thank you!

**Comment by tyhopp** [comment]
Hi @IhorFedorov7, I was not able to reproduce this. I recommend trying the command [listed in the starter itself](https://www.gatsbyjs.com/starters/gatsbyjs/gatsby-starter-blog/) in case there's some issue with your global install of gatsby cli:

```
npx gatsby new gatsby-starter-blog https://github.com/gatsbyjs/gatsby-starter-blog
```

**Reply by IhorFedorov7** [reply]
after installation:
npx gatsby new gatsby-starter-blog https://github.com/gatsbyjs/gatsby-starter-blog

And when running gatsby develop gives the error above
![image_2022-04-01_16-29-18 (3)](https://user-images.githubusercontent.com/50741511/161284371-dada8fd5-41cd-4515-a275-38af808fb7a0.png)

after npm install in the console is: 
![image_2022-04-01_16-29-18 (2)](https://user-images.githubusercontent.com/50741511/161284220-dd5a02b0-c7d9-4604-bc78-282663cef962.png)



And when you restart gatsby develop
![image_2022-04-01_16-29-18](https://user-images.githubusercontent.com/50741511/161284078-44aa4261-d3ad-4622-a37b-dcda2d58bc5a.png)



What I just did not try does not help.

what is most interesting, everything is ok on other equipment, I upload it to git, clone it to the problem device and the problem repeats.

**Comment by Primajin** [comment]
I am having the exact same issue when upgrading react to `18.0.0` and gatsby to `4.11.1`:
![image](https://user-images.githubusercontent.com/1742115/161419858-53549daa-64e3-4b8b-a847-d0d7e2384ebd.png)

I have not installed reach router or the likes as explicit dependency but can see that they are `required` from: `.cache/commonjs/production-app.js:11`

**Reply by Primajin** [reply]
I downgraded to react 17 until this is solved 😕  https://github.com/Primajin/eyesbound/pull/272

**Reply by MadaShindeInai** [reply]
Same issues with just updating gatsby from 4.8.0 version to 4.14.1

**Reply by MadaShindeInai** [reply]
And same issues with updating gatsby from 4.13.1 version to 4.14.1
UPD. current issue were solved when i **migrate from npm to yarn**

**Reply by Primajin** [reply]
I must admit ashamed that `gatsby clean` solved the issue 🤦🏻‍♂️

**Comment by barbalex** [comment]
I am having very similar problem in [this project](https://github.com/barbalex/vermehrung). Which happens every time I update @gatsbyjs/reach-router from v1.3.6 to 1.3.7, then run `yarn dev`:
```
success compile gatsby files - 2.003s
success load gatsby config - 0.029s
success load plugins - 0.952s
success onPreInit - 0.004s                                                                                                                                                                                                               
success initialize cache - 0.088s
success copy gatsby files - 0.242s
success Compiling Gatsby Functions - 0.233s
success onPreBootstrap - 0.255s
success createSchemaCustomization - 0.015s
success Checking for changed pages - 0.003s                                                                                                                                                                                              
success source and transform nodes - 0.292s                                                                                                                                                                                              
success building schema - 0.291s
success createPages - 0.044s
success createPagesStatefully - 0.100s
info Total nodes: 196, SitePage nodes: 38 (use --verbose for breakdown)                                                                                                                                                                  
success Checking for changed pages - 0.001s                                                                                                                                                                                              
success write out redirect data - 0.002s                                                                                                                                                                                                 
success Build manifest and related icons - 0.141s
success onPostBootstrap - 0.146s
info bootstrap finished - 6.812s                                                                                                                                                                                                         
success onPreExtractQueries - 0.001s                                                                                                                                                                                                     
success extract queries from components - 4.059s
success write out requires - 0.053s
success run static queries - 0.013s - 1/1 75.52/s                                                                                                                                                                                        
success run page queries - 0.039s - 3/3 77.13/s
success Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs - 0.702s - 4/4 5.69/s

 ERROR 

Module not found: Error: Can't resolve 'gatsby-react-router-scroll' in 'C:\Users\alexa\vermehrung\.cache'



  ModuleNotFoundError: Module not found: Error: Can't resolve 'gatsby-react-router-scroll' in 'C:\Users\alexa\vermehrung\.cache'

  - Compilation.js:2011
    [vermehrung]/[gatsby]/[webpack]/lib/Compilation.js:2011:28

  - NormalModuleFactory.js:795
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:795:13


  - NormalModuleFactory.js:275
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:275:22


  - NormalModuleFactory.js:431
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:431:22

  - NormalModuleFactory.js:124
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:124:11

  - NormalModuleFactory.js:667
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:667:25

  - NormalModuleFactory.js:852
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:852:8

  - NormalModuleFactory.js:972
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:972:5

  - async.js:6883
    [vermehrung]/[neo-async]/async.js:6883:13

  - NormalModuleFactory.js:955
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:955:45

  - Resolver.js:296 finishWithoutResolve
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:296:11

  - Resolver.js:362
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:362:15

  - Resolver.js:410
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:410:5



not finished Building development bundle - 4.220s   
```

**Reply by barbalex** [reply]
4 months later this is still consistently happening

**Comment by javiertury** [comment]
I was having the same problem. Turns out that `npm` installed `gatsby-react-router-scroll` and other dependencies in a nested path, like `node_modules/gatsby/node_modules/gatsby-react-router-scroll`, and webpack couldn't find them.

I was unable to make `npm` flatten the dependency tree, so I replaced it with pnpm. This is how I did it.

1. Create a `.npmrc` in your project with the following content

```
auto-install-peers=true
```

2. Remove old npm files and cache

```sh
rm -r package-lock.json node_modules .cache
```

3. Initialize pnpm

```sh
pnpm i
```

4. Add `gatsby-plugin-pnpm`

```sh
pnpm add -D gatsby-plugin-pnpm
```

5. Add the plugin to `gatsby-config.js`

```js
// gatsby-config.js
module.exports = {
  plugins: [
    `gatsby-plugin-pnpm`,
    ...
  ]
}
```

Now everything is working again

**Reply by BradNut** [reply]
@javiertury While this does work to resolve builds locally but depending on the CI/CD process you have it may not. In my case Netlify is not having it with pnpm builds (npx pnpm build, pnpm build || (npm install pnpm && pnpm build), repeat random search query suggestions for builds).

Any thought on getting it working with pnpm?

**Reply by javiertury** [reply]
I suspect there is something wrong with your CI/CD pipeline. Make sure to call `pnpm install` (replacement for `npm i`) or `pnpm install --frozen-lockfile` (replacement for` npm ci`) before running `pnpm run build`.

Do not use `npm i` or `npm ci`, they are not compatible with pnpm.

**Reply by BradNut** [reply]
@javiertury Alright I can try modifying the commands. At the moment I "resolved" the issue by installing all the packages that Gatsby complains it really wants and that works but isn't ideal.

**Reply by treckstar** [reply]
Thanks, I got this working with `pnpm` after getting a bunch of random dependency errors, all in the `.cache` folder. I still may have something else going on, based around `webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo` but the `pnpm run develop` at least works.

Using `gatsby ^4.24.7` and `react 17`

 Example of warnings:

> `<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Resolving 'lodash/package.json' in /var/www/aggressive-cache-blog/frontend/node_modules/.pnpm/babel-plugin-lodash@3.3.4/node_modules/babel-plugin-lodash/lib for build dependencies doesn't lead to expected result '/var/www/aggressive-cache-blog/node_modules/lodash/package.json', but to '/var/www/aggressive-cache-blog/frontend/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/package.json' instead. Resolving dependencies are ignored for this path.`
> `<w>  at unknown 4 lodash/package.json`

Thanks for sharing!

**Comment by barbalex** [comment]
This is annoying. Every go at updating dependencies is russian roulette. Whenever this error appears, I end up spending hours trying to get the project to work again. And this has been ongoing for over 4 months now.

The issue is so bad, I feel the only way to get rid of it is migrating away from gatsby. 

Why was this issue moved to a discussion?

**Reply by BradNut** [reply]
@barbalex How did you end up getting everything back up and running? I've had to do this same song and dance before as well but have no idea what actually fixes things.

Tried pnpm as suggested above which does work to resolve builds locally but depending on the CI/CD process you have it may not. In my case Netlify is not having it with pnpm builds / npx pnpm build etc.

**Reply by barbalex** [reply]
I reverted to previous yarn.lock file.

**Comment by tyhopp** [comment]
Hi all, so sorry you are experiencing lingering issues with this. I originally moved this to a discussion when it was probably better that I left it as an issue and requested for a minimal reproduction. Over time it's gathered various comments from people that have landed here, but it's difficult to separate signal from noise.

If you're still experiencing issues, please open a new issue with a [minimal reproduction](https://gatsby.dev/reproduction) and we'll do our best to triage. Thanks!

**Reply by barbalex** [reply]
This happens in only one of my projects. Which is basically huge. Sorry to say it is not possible for me to build a minimal reproduction. But yes, I spent again 5 hours today fighting this dragon. With no idea when or if the project will ever run again

**Comment by barbalex** [comment]
Today I updated @gatsbyjs/reach-router to 1.3.10-no-navigate.4 and gatsby 4.21.1 (including all the other gatsby stuff) and now get a slightly different error:

```
Module not found: Error: Can't resolve '@gatsbyjs/reach-router/lib/utils' in 'C:\Users\alexa\vermehrung\.cache'
```

So instead of `gatsby-react-router-scroll` it is missing `@gatsbyjs/reach-router/lib/utils` now.

Here the complete error:
```
failed extract queries from components - 4.225s                                                                                                                                                                                                           
success write out requires - 0.037s
success run static queries - 0.043s - 1/1 23.52/s
success run page queries - 0.043s - 3/3 69.15/s
success Running gatsby-plugin-sharp.IMAGE_PROCESSING jobs - 0.322s - 4/4 12.42/s

 ERROR 

Module not found: Error: Can't resolve '@gatsbyjs/reach-router/lib/utils' in 'C:\Users\alexa\vermehrung\.cache'



  ModuleNotFoundError: Module not found: Error: Can't resolve '@gatsbyjs/reach-router/lib/utils' in 'C:\Users\alexa\vermehrung\.cache'

  - Compilation.js:2011
    [vermehrung]/[gatsby]/[webpack]/lib/Compilation.js:2011:28

  - NormalModuleFactory.js:795
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:795:13


  - NormalModuleFactory.js:275
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:275:22


  - NormalModuleFactory.js:431
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:431:22

  - NormalModuleFactory.js:124
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:124:11

  - NormalModuleFactory.js:667
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:667:25

  - NormalModuleFactory.js:852
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:852:8

  - NormalModuleFactory.js:972
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:972:5

  - async.js:6883
    [vermehrung]/[neo-async]/async.js:6883:13

  - NormalModuleFactory.js:955
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:955:45

  - Resolver.js:296 finishWithoutResolve
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:296:11

  - Resolver.js:362
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:362:15

  - Resolver.js:410
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:410:5



                                                                                                                                                                                                                                                          
 ERROR 

Module not found: Error: Can't resolve 'gatsby-react-router-scroll' in 'C:\Users\alexa\vermehrung\.cache'



  ModuleNotFoundError: Module not found: Error: Can't resolve 'gatsby-react-router-scroll' in 'C:\Users\alexa\vermehrung\.cache'

  - Compilation.js:2011
    [vermehrung]/[gatsby]/[webpack]/lib/Compilation.js:2011:28

  - NormalModuleFactory.js:795
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:795:13


  - NormalModuleFactory.js:275
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:275:22


  - NormalModuleFactory.js:431
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:431:22

  - NormalModuleFactory.js:124
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:124:11

  - NormalModuleFactory.js:667
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:667:25

  - NormalModuleFactory.js:852
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:852:8

  - NormalModuleFactory.js:972
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:972:5

  - async.js:6883
    [vermehrung]/[neo-async]/async.js:6883:13

  - NormalModuleFactory.js:955
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:955:45

  - Resolver.js:296 finishWithoutResolve
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:296:11

  - Resolver.js:362
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:362:15

  - Resolver.js:410
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:410:5



                                                                                                                                                                                                                                                          
 ERROR 

Module not found: Error: Can't resolve 'gatsby-script' in 'C:\Users\alexa\vermehrung\.cache'



  ModuleNotFoundError: Module not found: Error: Can't resolve 'gatsby-script' in 'C:\Users\alexa\vermehrung\.cache'

  - Compilation.js:2011
    [vermehrung]/[gatsby]/[webpack]/lib/Compilation.js:2011:28

  - NormalModuleFactory.js:795
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:795:13


  - NormalModuleFactory.js:275
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:275:22


  - NormalModuleFactory.js:431
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:431:22

  - NormalModuleFactory.js:124
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:124:11

  - NormalModuleFactory.js:667
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:667:25

  - NormalModuleFactory.js:852
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:852:8

  - NormalModuleFactory.js:972
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:972:5

  - async.js:6883
    [vermehrung]/[neo-async]/async.js:6883:13

  - NormalModuleFactory.js:955
    [vermehrung]/[gatsby]/[webpack]/lib/NormalModuleFactory.js:955:45

  - Resolver.js:296 finishWithoutResolve
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:296:11

  - Resolver.js:362
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:362:15

  - Resolver.js:410
    [vermehrung]/[enhanced-resolve]/lib/Resolver.js:410:5



not finished Building development bundle - 9.673s       
```

I tried: 
- `yarn gatsby clean`: did not help
- `yarn gatsby clean;yarn add @gatsbyjs/reach-router@1.3.9;rm yarn.lock;rm -r node_modules;yarn;yarn dev`: Wow, that finally solved it 

I really do not know why I keep getting these errors almost every time I update gatsby related libraries.

**Reply by LekoArts** [reply]
You shouldn't need to install those dependencies separately, especially not versions that are not supposed to be used by end users like you. With `@gatsbyjs/reach-router@1.3.10-no-navigate.4` you've installed a canary version with breaking changes

**Reply by barbalex** [reply]
Seems that yarn upgrade-interactive lead me to install that version of reach-router without noticing.
But I then tried other versions and could not solve the issue.

**Comment by danielholmes** [comment]
Similar to https://github.com/gatsbyjs/gatsby/discussions/35313%22#discussioncomment-2494126 , downgrading to `react` + `react-dom` 17 worked for me.

My situation is pretty niche I think but I wanted to share in case it helps someone else. I'm using yarn workspaces (yarn version `1.22.19`) with the following packages:

 - website - a gatsby site
 - strapi - a strapi cms project
 
 The issue for me seemed to be that strapi uses react 17. So I was able to solve the problem by either:
 
  - Removing strapi dependencies from the strapi cms project (obviously not a viable solution, but done as an experiment)
  - Changing react + react-dom to 17 in the gatsby site project

**Comment by JustFly1984** [comment]
I'm getting issue running `yarn dev`:

```
ModuleNotFoundError: Module not found: Error: Can't resolve '@gatsbyjs/reach-router/lib/u
  tils' in '/Users/user/projects/personal/project/.cache'
```

before that I had to force install gatsby-scripts, gatsby-react-router-scroll, gatsby-link into package.json, I have `@gatsbyjs/reach-router` installed and new Head API refactored, but can't run dev, due an error above

It is just not looking down the node_modules tree. Please fix.

**Comment by JustFly1984** [comment]
Found the solution, described it in my latest issue: https://github.com/gatsbyjs/gatsby/issues/36565

**Comment by aaron5670** [comment]
Same problem here when we upgrade from React 17.0.2 to 18.2.
We are on Gatsby 4.22.1.

Any solution to this? Since there are several open issues on Github with the same problem.

**Comment by benpsnyder** [comment]
This worked for me in `gatsby-node.js`
```
/**
 * Implement Gatsby's Node APIs in this file.
 *
 * See: https://www.gatsbyjs.org/docs/node-apis/
 */

// You can delete this file if you're not using it

exports.onCreateWebpackConfig = ({
  stage,
  rules,
  loaders,
  plugins,
  actions,
}) => {
  actions.setWebpackConfig({
    resolve: {
      //modules: [...],
      fallback: {
        "zlib": require.resolve("browserify-zlib"),
        "url": require.resolve("url"),
        "os": require.resolve("os-browserify"),
        "http": require.resolve("stream-http"),
        "https": require.resolve("https-browserify"),
        "assert": require.resolve('assert'),
        "stream": require.resolve('stream-browserify'),
        "crypto": require.resolve('crypto-browserify'),
        "crypto-browserify": require.resolve('crypto-browserify'), //if you want to use this module also don't forget npm i crypto-browserify
      }
    },
  })
}
```
I use PNPM. `pnpm install -D crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url buffer process  browserify-zlib`

**Comment by beingraigari** [comment]
Please try installing the missing packages using npn install <package> --force. It should help.
